### DISCUSSION TITLE
Branch protection rules does not allow Github App

### DISCUSSION THREAD

**Discussion by awalach-aquiva** [community/community#77082] [Actions]
### Select Topic Area

Bug

### Body

Hey,
We have a repository with branch protection rules enabled. We need to auto-update some files when releasing a new version through Github Actions, so we wanted to bypass the rules in particular Github Actions Workflows using reco.
The app was created by the organization owner, and installed in the repository. Still, when I try to add it as an exception to the protection rule using the user interface, it returns an error "Could not resolve to a node with the global id of 'XXX'.".

The expected behaviour: either a human-readable error (need better plan? more permissions? I have no idea), or it should just work.

**Comment by DNin01** [comment]
Which feature are you using? Branch protection rules (`Settings` → `Branches`) or (the newer) rulesets (`Settings` → `Rules` → `Rulesets`)? Because if one of these is causing this issue for you, you might want to give the other one a try. Just an idea, not sure if it'd actually solve the problem. I've never used GitHub Apps!

**Comment by SacredSkull** [comment]
I can confirm this is still a problem. Why can you not assign github apps in the new(er) branch protection policy section? Do we really have to use classic protection rules in order to satisfy this?
